src/cryptonote_basic/cryptonote_format_utils.cpp

```diff
@@ -171,6 +183,13 @@ namespace cryptonote
     crypto::hash_to_scalar(data, sizeof(data), m);
     return m;
   }
+  crypto::secret_key get_subaddress_secret_key(const crypto::secret_key& a, const subaddress_index& index, hw::device &hwdev)
+  {
+    crypto::secret_key m;
+    hwdev.get_subaddress_secret_key(a, index, m);
+    return m;
+  }
```

Comment:
I have the same opinion as vtnerd, it would be better to keep the same prototype, just with an extra device parameter. But at this point I'm OK with leaving those if it's too much hassle.

Reply:
I will see. The waythe various subaddress functions are used and mixed make keeping the double prototype a bit hard and maybe dangerous. Need to think calmly about this.